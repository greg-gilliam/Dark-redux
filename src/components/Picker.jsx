import React from "react";

export default function Picker({
  head,
  onHeadChange,
  newCatchphrase,
  setCatchphrase,
  handleClick,
  body,
  onBodyChange,
  accessory,
  onAccessoryChange,
}) {
  // const [head, setHead] = useState("Gelfling");
  // const [body, setBody] = useState("Gelfling");
  // const [accessory, setAccessory] = useState("Shard");

  const headPics = ["Aughra", "Gelfling", "Skeksi", "urRu"];
  const bodyPics = ["Aughra", "Gelfling", "Skeksi", "urRu"];
  const accessoryPics = ["Shard", "Sword", "Fizzgig"];

  return (
    <div>
      <label>
        Head
        <select value={head} onChange={(e) => onHeadChange(e.target.value)}>
          {headPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>
      <label>
        Body
        <select value={body} onChange={(e) => onBodyChange(e.target.value)}>
          {bodyPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>
      <label>
        Accessorize!
        <select
          value={accessory}
          onChange={(e) => onAccessoryChange(e.target.value)}
        >
          {accessoryPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>
      <label>
        Add a catchphrase:
        <input
          type="text"
          value={newCatchphrase}
          onChange={(e) => setCatchphrase(e.target.value)}
        />
      </label>
      <button type="button" onClick={handleClick}>
        Add catchphrase to your list!
      </button>
    </div>
  );
}
