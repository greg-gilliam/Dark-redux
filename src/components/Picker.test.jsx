import { screen, render } from "@testing-library/react";
import Picker from "./Picker";

it("should render the Picker component", () => {
  const { container } = render(<Picker />);
  const selectHead = screen.getByLabelText(/Head/);
  const selectBody = screen.getByLabelText(/Body/);
  const selectAccessory = screen.getByLabelText(/Accessorize!/);
  const input = screen.getByLabelText(/Add a catchphrase:/);

  expect(selectHead).toBeInTheDocument();
  expect(selectBody).toBeInTheDocument();
  expect(selectAccessory).toBeInTheDocument();
  expect(input).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
