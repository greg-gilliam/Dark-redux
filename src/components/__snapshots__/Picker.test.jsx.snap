// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Picker component 1`] = `
<div>
  <div>
    <label>
      Head
      <select>
        <option>
          Aughra
        </option>
        <option>
          Gelfling
        </option>
        <option>
          Skeksi
        </option>
        <option>
          urRu
        </option>
      </select>
    </label>
    <label>
      Body
      <select>
        <option>
          Aughra
        </option>
        <option>
          Gelfling
        </option>
        <option>
          Skeksi
        </option>
        <option>
          urRu
        </option>
      </select>
    </label>
    <label>
      Accessorize!
      <select>
        <option>
          Shard
        </option>
        <option>
          Sword
        </option>
        <option>
          Fizzgig
        </option>
      </select>
    </label>
    <label>
      Add a catchphrase:
      <input
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Add catchphrase to your list!
    </button>
  </div>
</div>
`;
